File: snake.py
import random
import os
import time
import msvcrt

width, height = 20, 10
snake = [(5, 5)]
direction = (1, 0)
food = (random.randint(1, width), random.randint(1, height))
game_over = False

def clear_screen():
    os.system('cls' if os.name == 'nt' else 'clear')

def draw():
    clear_screen()
    for y in range(1, height + 1):
        for x in range(1, width + 1):
            if (x, y) in snake:
                print("O", end="")
            elif food == (x, y):
                print("X", end="")
            else:
                print(".", end="")
        print()

def update():
    global game_over, food
    new_head = (snake[0][0] + direction[0], snake[0][1] + direction[1])

    if (new_head[0] < 1 or new_head[0] > width or
        new_head[1] < 1 or new_head[1] > height or
        new_head in snake):
        game_over = True
        return

    snake.insert(0, new_head)

    if new_head == food:
        food = (random.randint(1, width), random.randint(1, height))
    else:
        snake.pop()

def change_direction(new_direction):
    global direction
    if (new_direction[0] == -direction[0] and
            new_direction[1] == -direction[1]):
        return
    direction = new_direction

while not game_over:
    draw()
    update()

    if msvcrt.kbhit():
        key = msvcrt.getch()
        if key == b'w':
            change_direction((0, -1))
        elif key == b's':
            change_direction((0, 1))
        elif key == b'a':
            change_direction((-1, 0))
        elif key == b'd':
            change_direction((1, 0))

    time.sleep(0.1)

print("Game Over!")

File: snake.cs
using System;
using System.Collections.Generic;
using System.Threading;

class Program
{
    static int width = 20, height = 10;
    static List<(int x, int y)> snake = new List<(int, int)> { (5, 5) };
    static (int x, int y) direction = (1, 0);
    static (int x, int y) food = (new Random().Next(1, width + 1), new Random().Next(1, height + 1));
    static bool gameOver = false;

    static void ClearScreen()
    {
        Console.Clear();
    }

    static void Draw()
    {
        ClearScreen();
        for (int y = 1; y <= height; y++)
        {
            for (int x = 1; x <= width; x++)
            {
                if (snake.Contains((x, y)))
                    Console.Write("O");
                else if (food == (x, y))
                    Console.Write("X");
                else
                    Console.Write(".");
            }
            Console.WriteLine();
        }
    }

    static void Update()
    {
        var newHead = (snake[0].x + direction.x, snake[0].y + direction.y);

        if (newHead.x < 1 || newHead.x > width || newHead.y < 1 || newHead.y > height || snake.Contains(newHead))
        {
            gameOver = true;
            return;
        }

        snake.Insert(0, newHead);

        if (newHead == food)
        {
            food = (new Random().Next(1, width + 1), new Random().Next(1, height + 1));
        }
        else
        {
            snake.RemoveAt(snake.Count - 1);
        }
    }

    static void ChangeDirection((int x, int y) newDirection)
    {
        if (newDirection.x == -direction.x && newDirection.y == -direction.y) return;
        direction = newDirection;
    }

    static void Main()
    {
        while (!gameOver)
        {
            Draw();
            Update();

            if (Console.KeyAvailable)
            {
                var key = Console.ReadKey(true).Key;
                switch (key)
                {
                    case ConsoleKey.W: ChangeDirection((0, -1)); break;
                    case ConsoleKey.S: ChangeDirection((0, 1)); break;
                    case ConsoleKey.A: ChangeDirection((-1, 0)); break;
                    case ConsoleKey.D: ChangeDirection((1, 0)); break;
                }
            }

            Thread.Sleep(100);
        }

        Console.WriteLine("Game Over!");
    }
}
